.github/workflows/trivy-cache.yml

```diff
@@ -0,0 +1,45 @@
+---
+name: Cache Trivy DBs
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+
+permissions:
+  packages: write
+  id-token: write
+
+jobs:
+  caching:
+    name: Cache Trivy DBs
+    runs-on: ubuntu-latest
+    env:
+      repo_name: gsa-tts/fac
+    steps:
+      - name: Install ORAS
+        id: oras
+        uses: oras-project/setup-oras@v1
+
+      - name: Authenticate to GHCR
+        id: ghcr
+        run: |
+          oras login ghcr.io \
+            -u ${{ github.actor }} \
+            -p ${{ github.token }}
+
+      - name: Pull Trivy Images
+        id: pull
+        run: |
+          oras pull ghcr.io/aquasecurity/trivy-db:2
+          oras pull ghcr.io/aquasecurity/trivy-java-db:1
+

```

Comment:
Do we need a step here to create tarballs first before pushing them ?

Reply:
On push, it is pushing the compressed layer
`db.tar.gz:application/vnd.aquasec.trivy.db.layer.v1.tar+gzip`

Reply:
This being said, a few others seem to not be building their own, and just adding a secondary link which appears to slow the ratelimiting
`TRIVY_DB_REPOSITORY: ghcr.io/aquasecurity/trivy-db,public.ecr.aws/aquasecurity/trivy-db`

Reply:
Modified workflow to store in cache nightly but added the extra trivy location to hopefully help minimize issues if it does have to download